portalnet/src/overlay_service.rs

```diff
@@ -1432,25 +1447,30 @@ where
 
             // send the content to the acceptor over a uTP stream
             if let Err(err) = stream.write(&content_payload).await {
+                metrics.report_outbound_utp_tx(&protocol, false);
                 warn!(
                     %err,
                     cid.send,
                     cid.recv,
                     peer = ?cid.peer.client(),
                     "Error sending content over uTP connection"
                 );
+                return;
```

Comment:
I added a `return` here since I'd like to avoid falsely reporting two failures for a single tx should `stream.shutdown()` also fail. But, I'm not sure if that's ideal behavior, and we still want to explicitly shutdown the stream even if the tx failed. @jacobkaufmann ?

Reply:
I would get rid of the return and just use a local `mut bool` to make sure you only report once.
```
let mut reported_stream_failure: bool = false;

...

if !reported_stream_failure { 
     metrics.report_outbound_utp_tx(&protocol, false);
     reported_stream_failure = true;
 }
```

I'm assuming that calling `shutdown` after a failed write is intentional and correct, if it's not then I agree with your `return`.

Reply:
I think we still want to shut down the stream even if the write fails. I would remove the return.